### ISSUE TITLE
Add some sort of symlink management for indirect backups

### ISSUE THREAD

**Issue by mtkennerly** [mtkennerly/ludusavi#11]
GSM has a "Steam Spreader" function for moving/symlinking game installs, as well as a "Link & Sync" function for doing the same just for save data. It may be worth adding similar functionality here for users coming from GSM, since it provides a way to redirect game saves to a cloud storage folder or alternative drive, thus acting as another form of backup. This would require some care to ensure it works well cross-platform and that it won't break anything (e.g., if folders/files are affected that shouldn't be or if it could interfere with the operation of Steam/etc).

**Comment by Triplesweet** [comment]
I tried the feature for a day and this is why I like to say, having this feature for game files makes no sense since you can move them from steam easier imo, for save files I feel making the back up location in ludusavi a linked folder (to any service) makes it easier, it is true that it will be all automatic,

But here is the point, having game play backup and backing up the files on a synced folder is much easier than or maybe some kind of tracker that looks at the active exe files, and backed up that game each 10 minutes or so, or user can decide, it feels redundant If I have to be honest.  

In short having steam spreader feels a waste of time, and sync and link feature can be considered redundant and gameplay backup can be modified to be better.

These are all my opinions after using it only for a day, maybe people found some cool ideas to use them, but then again I'm just an idiot ¯⁠\⁠_⁠(⁠ツ⁠)⁠_⁠/⁠¯

**Comment by mtkennerly** [comment]
Thanks for your feedback. I personally agree, and I do have some other reservations around this feature.

I think it would be hard to get right, and the cost of failure is high since it could corrupt save data. I don't want it to turn into a footgun, and I don't think it really fits the design goals of Ludusavi. Some specific challenges:

* Behavior from Steam and other clients could be unpredictable. What works today might not work tomorrow, and individual games might have trouble handling unexpected symlinks. I never want Ludusavi to corrupt people's save data by accident.
* Currently, Ludusavi backs up the target of symlinks, but not the symlinks themselves. If Ludusavi were also responsible for creating symlinks, it would have to back those up in a different way from "unmanaged" symlinks. However, that means if you cleared out your Ludusavi config (e.g., reinstall or even running as a different user), then it may not be able to back up/restore them properly. Someone ran into a similar issue with GSM: https://forum.gamesave-manager.com/viewtopic.php?p=716

For anyone interested in this, you could still use Ludusavi's CLI (with `--api`) to automate something that reports/symlinks files that exceed a certain size. I think this is a good opportunity for another script/tool to build on top of Ludusavi.
